### PR TITLE
DevEx: use docketNumber instead of caseId on Case Deadlines

### PR DESCRIPTION
**Pull request by rachaelparris**

### CODE REVIEW

**Review by matthopson — APPROVED** [review]
Do the Proxy updates need to be updated in swagger?

**Comment by rachaelparris** [comment]
> Do the Proxy updates need to be updated in swagger?

Yes, good catch!

**Review by matthopson — APPROVED** [review]

File: shared/src/business/useCases/getAllCaseDeadlinesInteractor.js [hunk 1]
```diff
@@ -35,32 +35,32 @@ exports.getAllCaseDeadlinesInteractor = async ({ applicationContext }) => {
   );
 
   // get the needed cases info data for caseDeadlines
-  const caseIds = Object.keys(
+  const docketNumbers = Object.keys(
     validatedCaseDeadlines.reduce((acc, item) => {
-      acc[item.caseId] = true;
+      acc[item.docketNumbers] = true;
```
Comment (kkoskelin) [inline shared/src/business/useCases/getAllCaseDeadlinesInteractor.js]: * `docketNumber`?

File: shared/src/business/useCases/getAllCaseDeadlinesInteractor.js [hunk 2]
```diff
@@ -35,32 +35,32 @@ exports.getAllCaseDeadlinesInteractor = async ({ applicationContext }) => {
...
+      docketNumbers,
     });
 
   const validatedCaseData = Case.validateRawCollection(allCaseData, {
     applicationContext,
   });
 
   const caseMap = validatedCaseData.reduce((acc, item) => {
-    acc[item.caseId] = item;
+    acc[item.docketNumbers] = item;
```
Comment (kkoskelin) [inline shared/src/business/useCases/getAllCaseDeadlinesInteractor.js]: *`docketNumber`?

File: shared/src/business/useCases/getAllCaseDeadlinesInteractor.js [hunk 3]
```diff
@@ -35,32 +35,32 @@ exports.getAllCaseDeadlinesInteractor = async ({ applicationContext }) => {
...
     return acc;
   }, {});
 
   const afterCaseMapping = validatedCaseDeadlines.map(deadline => ({
     ...deadline,
-    ...pick(caseMap[deadline.caseId], [
+    ...pick(caseMap[deadline.docketNumbers], [
```
Comment (kkoskelin) [inline shared/src/business/useCases/getAllCaseDeadlinesInteractor.js]: *`docketNumber`?

File: web-api/src/caseDeadline/deleteCaseDeadlineLambda.js [hunk 4]
```diff
@@ -11,6 +11,6 @@ exports.deleteCaseDeadlineLambda = event =>
     return await applicationContext.getUseCases().deleteCaseDeadlineInteractor({
       applicationContext,
       caseDeadlineId: event.pathParameters.caseDeadlineId,
-      caseId: event.pathParameters.caseId,
+      docketNumber: event.pathParameters.docketNumber,
```
Comment (kkoskelin) [inline web-api/src/caseDeadline/deleteCaseDeadlineLambda.js]: Personally I prefer `...event.pathParameters,` so we don't have to call out specific properties in the lambda.

**Review by kkoskelin — DISMISSED** [review]
Some things are def. bugs, others are just nits. you can tell which are which, I bet. :D

**Pull request merged**